compiler/passes/scopeResolve.cpp

```diff
@@ -1043,6 +1043,12 @@ static void build_constructor(AggregateType* ct) {
       if (method->numFormals() == 2) {
         CallExpr* init = new CallExpr("initialize", gMethodToken, fn->_this);
         fn->insertAtTail(init);
+        // If a type has an initialize method, it's not Plain Old Data.
+        // The only time classes aren't Plain Old Data is when they have the
+        // pragma "no object" associated with them.
+        if (!isClass(ct) || ct->symbol->hasFlag(FLAG_NO_OBJECT)) {
+          ct->symbol->addFlag(FLAG_NOT_POD);
+        }
         break;
```

Comment:
Again, I am OK with this going in, however I do not see the relationship of having-object-parent to POD-ness. Please add a comment to the code explaining the relationship.

Furthermore, if a class type can be POD or not, it means (a) we can allow user assignment overloads, and (b) we need to track POD-ness of classes consistently throughout.

Towards that end, I suggest that each Type symbol ends up with exactly one of FLAG_POD and FLAG_NOT_POD coming out of resolution.